sdk/python/feast/feature_set.py

```diff
@@ -663,6 +660,41 @@ def is_valid(self):
         if len(self.entities) == 0:
             raise ValueError(f"No entities found in feature set {self.name}")
 
+    def update_schema(self, schema: Schema):
+        """
+        Updates presence_constraints, shape_type and domain_info for all entities
+        and features in the FeatureSet from schema in Tensorflow metadata.
+
+        Args:
+            schema: schema from Tensorflow metadata
+
+        Returns:
+            None
+
+        """
+        name_to_feature = {f.name: f for f in self.features}
+        name_to_entity = {e.name: e for e in self.entities}
```

Comment:
Can this be deduped?
`name_to_field = {f.name: f for f in self.features + self.entities}`

Reply:
Ok